plugins/single_plugins/vswitch.cpp

```diff
@@ -197,12 +197,18 @@ class vswitch : public wf::plugin_interface_t
 
     wf::signal_connection_t on_set_workspace_request = [=] (wf::signal_data_t *data)
     {
+        auto ev = static_cast<wf::workspace_change_request_signal*>(data);
         if (is_active())
         {
-            return;
+            auto cws = output->workspace->get_current_workspace();
```

Comment:
Does this fix it at all? I'd think not.

Reply:
Yes, at least in the case I could reproduce this issue before. it seems that cws does not change until the animation is finished. But this might not always be the case?
That's why I wrote I'm unsure about the ev->old_viewport parameter as well

Reply:
Oh, I realized a problem here: add_direction should not be called in this case; I wonder why it was still working

Reply:
`add_direction` is always the correct thing to do. The new direction would be `ev->new_viewport.x - animation.dx.end`

Reply:
I found that it still needs to take into consideration the old workspace; I've added an updated version